socketio/src/client/client.rs

```diff
@@ -524,6 +535,37 @@ mod test {
         test_socketio_socket(socket, "/admin".to_owned())
     }
 
+    #[test]
+    fn socket_io_auth_builder_integration() -> Result<()> {
+        let url = crate::test::socket_io_auth_server();
+        let socket = ClientBuilder::new(url)
+            .namespace("/admin")
+            .auth(json!({ "password": "123" }).to_string())
+            .connect()?;
+
+        let (tx, rx) = mpsc::sync_channel(0);
+
+        assert!(socket
+            .emit_with_ack(
+                "test",
+                json!({ "msg": "1" }),
+                Duration::from_secs(1),
+                move |payload, _| {
+                    println!("Got ack");
+                    tx.send(Payload::String(json!(["456"]).to_string()) == payload)
+                        .unwrap();
+                }
+            )
+            .is_ok());
+
+        sleep(Duration::from_secs(2));
+
+        let received = rx.recv();
```

Comment:
The `rx.recv()` call blocks, why do we need to sleep beforehand?

Reply:
I just forgot to delete that line :D